src/GraphQL/GraphQLExtensions.cs

```diff
@@ -279,16 +279,15 @@ public static string[] IsValidLiteralValue(this IGraphType type, IValue valueAst
                 return errors.ToArray();
             }
 
-            var scalar = (ScalarGraphType)type;
-
-            var parseResult = scalar.ParseLiteral(valueAst);
-
-            if (parseResult == null)
+            if (type is ScalarGraphType scalar)
             {
-                return new[] { $"Expected type \"{type.Name}\", found {AstPrinter.Print(valueAst)}." };
+                return scalar.CanParseLiteral(valueAst)
+                    ? Array.Empty<string>()
+                    : new[] { $"Expected type \"{type.Name}\", found {AstPrinter.Print(valueAst)}." };
             }
 
-            return Array.Empty<string>();
+            // the code shouldn't go here
+            return new[] { $"Value {AstPrinter.Print(valueAst)} of unknown type {type?.Name}." };
```

Comment:
Would this be more appropriate?

```suggestion
            throw new ArgumentOutOfRangeException(nameof(type), $"Type {type?.Name} is not a valid input graph type.");
```